src/plugins/clips-executive/clips/wm-robmem-sync.clp

```diff
@@ -98,15 +98,12 @@
 )
 
 (deffunction wm-robmem-sync-restore-prefix (?id-prefix)
-	(bind ?regex (bson-create))
-	(bson-append ?regex "$regex" (str-cat "^" ?id-prefix))
 	(bind ?query (bson-create))
-	(bson-append ?query "_id" ?regex)
-	(bson-destroy ?regex)
+	(bson-append-regex ?query "_id" (str-cat "^" ?id-prefix ))
 	(bind ?cursor (robmem-query ?*WM-ROBMEM-SYNC-COLLECTION* ?query))
 	(if ?cursor then
-		(while (robmem-cursor-more ?cursor) do
-			(bind ?doc (robmem-cursor-next ?cursor))
+		(bind ?doc (robmem-cursor-next ?cursor))
+    (while (neq FALSE ?doc) do
```

Comment:
I guess the following should work:
```suggestion
    (while ?doc do
```

Also, please fix the indentation.

Reply:
You are right. Fixed